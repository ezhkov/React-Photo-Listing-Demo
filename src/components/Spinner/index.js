import React, { memo } from 'react';
import './Spinner.css';

const Spinner = memo(() => (
  <div className="site-spinner">
    <div className="lds-ring">
      <div>&nbsp;</div>
      <div>&nbsp;</div>
      <div>&nbsp;</div>
      <div>&nbsp;</div>
    </div>
  </div>
));

export default Spinner;
