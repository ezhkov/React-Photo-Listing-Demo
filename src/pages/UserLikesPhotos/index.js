import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Row, Col, Pagination } from 'antd';
import { userLikesRequestAction } from '../../actions';
import { Spinner, PhotoCard, Error } from '../../components';
import './index.css';

class UserLikesPhotos extends PureComponent {

  componentDidMount = () => {
    const {
      userId,
      page,
      perPage,
      userLikesRequestAction: requestAction,
    } = this.props;

    requestAction(userId, page, perPage);
  };

  handlePaginationChange = (current) => {
    const {
      userId,
      perPage,
      userLikesRequestAction: requestAction,
    } = this.props;

    requestAction(userId, current, perPage);
  };

  render() {
    const {
      isUserLikesFetching,
      cards,
      totalCards,
      page,
      perPage,
      requestError,
    } = this.props;

    return (
      <div>
        { isUserLikesFetching && (<Spinner />)}
        { !isUserLikesFetching && !requestError && (
        <div>
          <Row justify="center" style={{ margin: '20px 0' }}>
            <Col span={24}>
              {!isUserLikesFetching && (
                <ul className="photo-list user-photos">
                  {cards.map(item => (
                    <li
                      key={item.photoID}
                      className="photo-list__item pl-3"
                    >
                      <PhotoCard
                        photoName={item.photoName}
                        title={item.title}
                        tags={item.tags}
                        photoID={item.photoID}
                        userID={item.userID}
                        userAvatar={item.userAvatar}
                        onSearchTagValue={this.handleSearchText}
                      />
                    </li>
                  ))
                  }
                </ul>
              )}
            </Col>
          </Row>
          <Row
            justify="center"
            style={{
              display: 'flex',
              justifyContent: 'center',
            }}
          >
            <Col style={{
              display: 'flex',
              justifyContent: 'center',
            }}
            >
              {totalCards > perPage && (
                <Pagination
                  className="ml-3 mb-5"
                  onChange={this.handlePaginationChange}
                  hideOnSinglePage
                  current={page}
                  defaultCurrent={1}
                  total={totalCards}
                  pageSize={6}
                />
              )}
            </Col>
          </Row>
        </div>
        )}
        { !isUserLikesFetching && requestError && (
        <Error text="Sorry, an error occurred during the request. Try again later." />
        )}
      </div>
    );
  }
}

UserLikesPhotos.propTypes = {
  isUserLikesFetching: PropTypes.bool,
  cards: PropTypes.arrayOf(PropTypes.shape({})),
  totalCards: PropTypes.number,
  page: PropTypes.number,
  perPage: PropTypes.number,
  userLikesRequestAction: PropTypes.func,
  requestError: PropTypes.bool,
  history: PropTypes.shape({
    prop: PropTypes.string,
  }),
  userId: PropTypes.string,
};
UserLikesPhotos.defaultProps = {
  isUserLikesFetching: true,
  cards: [],
  totalCards: 10,
  page: 1,
  perPage: 6,
  userLikesRequestAction: () => {},
  requestError: false,
  history: {},
  userId: '',
};
const mapStateToProps = (state) => {
  const { userlikesphotos } = state;
  return userlikesphotos;
};

const mapDispatchToProps = ({
  userLikesRequestAction,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(UserLikesPhotos);
