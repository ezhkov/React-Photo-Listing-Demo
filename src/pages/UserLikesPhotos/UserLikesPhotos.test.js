import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UserLikesPhotos from '.';
import Spinner from '../../components/Spinner';

Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of UserLikesPhotos', () => {
  it('Test display Pagination', () => {
    const props = {
      userId: 'sC-BXbi9ajw',
    };

    const userLikesPhotos = shallow(<UserLikesPhotos {...props} />);

    userLikesPhotos.setState({
      totalCards: 7,
    });

    expect(userLikesPhotos.find('Pagination').length).toEqual(1);

    userLikesPhotos.setState({
      totalCards: 5,
    });

    expect(userLikesPhotos.find('Pagination').length).toEqual(0);
  });

  it('Test Spinner and photo-list', () => {
    const props = {
      userId: 'sC-BXbi9ajw',
    };

    const userLikesPhotos = shallow(<UserLikesPhotos {...props} />);


    userLikesPhotos.setState({
      isListingLoading: false,
    });
    expect(userLikesPhotos.find(Spinner).length).toEqual(0);
    expect(userLikesPhotos.find('.user-photos').length).toEqual(1);


    userLikesPhotos.setState({
      isListingLoading: true,
    });

    expect(userLikesPhotos.find(Spinner).length).toEqual(1);
    expect(userLikesPhotos.find('.user-photos').length).toEqual(0);
  });

  it('Test Snapshot', () => {
    const props = {
      userId: 'sC-BXbi9ajw',
    };

    const userLikesPhotos = shallow(<UserLikesPhotos {...props} />);
    expect(userLikesPhotos).toMatchSnapshot();
  });
});
