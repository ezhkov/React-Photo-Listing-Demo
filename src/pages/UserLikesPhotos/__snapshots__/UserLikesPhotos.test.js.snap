// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of UserLikesPhotos Test Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserLikesPhotos
    userId="sC-BXbi9ajw"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Spinner />,
        <div>
          <Row
            gutter={0}
            justify="center"
            style={
              Object {
                "margin": "20px 0",
              }
            }
          >
            <Col
              span={24}
            />
          </Row>
          <Row
            gutter={0}
            justify="center"
            style={
              Object {
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <Col
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Pagination
                className="ml-3 mb-5"
                current={1}
                defaultCurrent={1}
                onChange={[Function]}
                prefixCls="ant-pagination"
                selectPrefixCls="ant-select"
                showSizeChanger={true}
                total={10}
              />
            </Col>
          </Row>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Row
              gutter={0}
              justify="center"
              style={
                Object {
                  "margin": "20px 0",
                }
              }
            >
              <Col
                span={24}
              />
            </Row>,
            <Row
              gutter={0}
              justify="center"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Col
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <Pagination
                  className="ml-3 mb-5"
                  current={1}
                  defaultCurrent={1}
                  onChange={[Function]}
                  prefixCls="ant-pagination"
                  selectPrefixCls="ant-select"
                  showSizeChanger={true}
                  total={10}
                />
              </Col>
            </Row>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Col
                span={24}
              />,
              "gutter": 0,
              "justify": "center",
              "style": Object {
                "margin": "20px 0",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": false,
                "span": 24,
              },
              "ref": null,
              "rendered": false,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Col
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <Pagination
                  className="ml-3 mb-5"
                  current={1}
                  defaultCurrent={1}
                  onChange={[Function]}
                  prefixCls="ant-pagination"
                  selectPrefixCls="ant-select"
                  showSizeChanger={true}
                  total={10}
                />
              </Col>,
              "gutter": 0,
              "justify": "center",
              "style": Object {
                "display": "flex",
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Pagination
                  className="ml-3 mb-5"
                  current={1}
                  defaultCurrent={1}
                  onChange={[Function]}
                  prefixCls="ant-pagination"
                  selectPrefixCls="ant-select"
                  showSizeChanger={true}
                  total={10}
                />,
                "style": Object {
                  "display": "flex",
                  "justifyContent": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "ml-3 mb-5",
                  "current": 1,
                  "defaultCurrent": 1,
                  "onChange": [Function],
                  "prefixCls": "ant-pagination",
                  "selectPrefixCls": "ant-select",
                  "showSizeChanger": true,
                  "total": 10,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Spinner />,
          <div>
            <Row
              gutter={0}
              justify="center"
              style={
                Object {
                  "margin": "20px 0",
                }
              }
            >
              <Col
                span={24}
              />
            </Row>
            <Row
              gutter={0}
              justify="center"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Col
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <Pagination
                  className="ml-3 mb-5"
                  current={1}
                  defaultCurrent={1}
                  onChange={[Function]}
                  prefixCls="ant-pagination"
                  selectPrefixCls="ant-select"
                  showSizeChanger={true}
                  total={10}
                />
              </Col>
            </Row>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Row
                gutter={0}
                justify="center"
                style={
                  Object {
                    "margin": "20px 0",
                  }
                }
              >
                <Col
                  span={24}
                />
              </Row>,
              <Row
                gutter={0}
                justify="center"
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <Col
                  style={
                    Object {
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Pagination
                    className="ml-3 mb-5"
                    current={1}
                    defaultCurrent={1}
                    onChange={[Function]}
                    prefixCls="ant-pagination"
                    selectPrefixCls="ant-select"
                    showSizeChanger={true}
                    total={10}
                  />
                </Col>
              </Row>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Col
                  span={24}
                />,
                "gutter": 0,
                "justify": "center",
                "style": Object {
                  "margin": "20px 0",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": false,
                  "span": 24,
                },
                "ref": null,
                "rendered": false,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Col
                  style={
                    Object {
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Pagination
                    className="ml-3 mb-5"
                    current={1}
                    defaultCurrent={1}
                    onChange={[Function]}
                    prefixCls="ant-pagination"
                    selectPrefixCls="ant-select"
                    showSizeChanger={true}
                    total={10}
                  />
                </Col>,
                "gutter": 0,
                "justify": "center",
                "style": Object {
                  "display": "flex",
                  "justifyContent": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Pagination
                    className="ml-3 mb-5"
                    current={1}
                    defaultCurrent={1}
                    onChange={[Function]}
                    prefixCls="ant-pagination"
                    selectPrefixCls="ant-select"
                    showSizeChanger={true}
                    total={10}
                  />,
                  "style": Object {
                    "display": "flex",
                    "justifyContent": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "ml-3 mb-5",
                    "current": 1,
                    "defaultCurrent": 1,
                    "onChange": [Function],
                    "prefixCls": "ant-pagination",
                    "selectPrefixCls": "ant-select",
                    "showSizeChanger": true,
                    "total": 10,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
